lib/async_hooks.js

```diff
@@ -351,6 +353,49 @@ class AsyncLocalStorage {
       return resource[this.kResourceStore];
     }
   }
+
+  bindToTracingChannel(channel, transform = (v) => v) {
+    if (typeof channel === 'string') {
+      channel = tracingChannel(channel);
+    }
+
+    const store = this;
+
+    function start(ctx) {
+      if (store.enabled) {
+        ctx[store.kResourceStore] = store.getStore();
+      }
+      store.enterWith(transform(ctx.data));
+    }
+
+    function end(ctx) {
+      if (ObjectPrototypeHasOwnProperty(ctx, store.kResourceStore)) {
+        store.enterWith(ctx[store.kResourceStore]);
+      } else {
+        store.disable();
+      }
+    }
+
+    channel.subscribe({
+      start,
+      end
+    });
+
+    return () => {
+      channel.unsubscribe({
+        start,
+        end
+      });
+    };
+  }
+
+  static runInTracingChannel(channel, data, runner) {
+    if (typeof channel === 'string') {
+      channel = tracingChannel(channel);
+    }
+
+    channel.traceSync(runner, { data });
```

Comment:
I took me a while to find the transformation to move the published start data into the `.data` property which is used in `start()` closure to call `transform`.

So it seems like a publisher should use this API from `AsyncLocalStore` to publish to ensure its `TracingChannel` fits to `AsyncLocalStore`. I know it's not strictly required as creating a context object with a `data` property can be also done in producer code.

But isn't the target to decouple the producing side using `TracingChannel` and consuming side subscribing and maybe bind to `AsyncLocalStore`?